packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -161,6 +162,10 @@ const DATA_MAP: AuthDataMap = {
     dwsid: {
         storageType: 'cookie',
         key: 'dwsid'
+    },
+    code_verifier: {
+        storageType: 'cookie',
+        key: 'code_verifier'
```

Comment:
Because `authorizeCustomer` and `getAccessToken` are in separate `isomorphic` helpers, there needs to be a way to transfer the `codeVerifier` from the first API call to the second. Does storing this as a cookie pose a security risk?

Reply:
@kevinxh would be the SME for this. Lets see if he has any recommendation. 

As a minimum I don't see why we couldn't use the 'local' storage type here and immediately delete it after using it.

Reply:
Cookie is meant to be persistent, but in your case, you only need to store it in the lifecycle of the javascript program, so memory is a better choice. 

I would actually suggest don't add `code_verifier` to the `DATA_MAP` and store it in the `auth` instance as a property, i.e. `this.code_verifier`, because this is a property that we never want to store in a persistent storage because that would leak it across context. (imagine two requests from two different user clicks loginIDP at the same time, if `code_verifier` is shared in a persistent storage, then there is a chance the second request uses the `code_verifier` from the first request.), keeping it as a property of the `Auth` class makes sure that this property is only available in the context of a single request.

Reply:
next question: why do we want to have two helpers `authorizeIDP` and `loginIDPUser`, and how would them being used in a real use case?

If you have only one helper, then you don't need to worry about sharing state and it might be easier to use, DX wise.

Reply:
@kevinxh [Here](https://github.com/SalesforceCommerceCloud/commerce-sdk-isomorphic/pull/172) is the link to the `commerce-sdk-isomorphic` PR that implemented the social login helpers.

In social login, the SLAS `authorizeCustomer` endpoint responds with a `303` that should redirect the user to the 3rd party IDP login page. However, when calling it inside the `isomorphic` repo with `fetch`, there were a lot of CORS issues in dealing with the redirects, and the existing `authorize` helper didn't return the `code` and `usid` properly, which is needed for the following, 2nd API call (`getAccessToken`) in the login flow. Thus, I decided to separate the 2 calls, so that `authorizeIDP` returns the request URL - I found that calling this endpoint from the top-level browser redirects the user properly to the login page in a separate tab and back to the storefront after successful login.

Reply:
Therefore the social login flow wasn't able to be combined into a single login helper.

This is the same for passwordless login - we separated the 2 calls because we need to wait for the user to either input a token or click a magic link that will take them back to the app to complete the login flow.

Reply:
ah, ok, I get it now, I was thinking from the server's perspective but this is a client side action.